app/attributes/attributecontroller.cpp

```diff
@@ -249,8 +252,39 @@ void AttributeController::flatten(
       }
 
       case QgsAttributeEditorElement::AeTypeRelation:
-        // todo
+      {
+        QUuid widgetUuid = QUuid::createUuid();
+
+        QgsAttributeEditorRelation *relationField = static_cast<QgsAttributeEditorRelation *>( element );
+        QgsRelation relation = relationField->relation();
```

Comment:
GitHub highlight "relation" as c++ keyword (there is std::relation..) so maybe use some other name for local variable?

Reply:
It will still highlight the `->relation` part